### PR TITLE
Update `UNet2DConditionModel`'s error messages

### PR DESCRIPTION
**Pull request by tolgacangoz**
I have been implementing Matryoshka Diffusion Models. While reviewing the UNet-related files, I wanted to update some of the error messages and remove seemingly unnecessary lines.

@DN6 @yiyixuxu @sayakpaul

### CODE REVIEW

File: src/diffusers/models/unets/unet_2d_condition.py [hunk 1]
```diff
@@ -463,7 +463,6 @@ def __init__(
                 dropout=dropout,
             )
             self.up_blocks.append(up_block)
-            prev_output_channel = output_channel
```
Comment (tolgacangoz) [inline src/diffusers/models/unets/unet_2d_condition.py:466]: This line seems unnecessary. Am I right here?

**Comment by tolgacangoz** [comment]
@ellipsis-dev Hey, review pls!

**Comment by tolgacangoz** [comment]
@ellipsis-dev review this

File: src/diffusers/models/unets/unet_2d_condition.py (hunk 1 above)
Comment (tolgacangoz) [reply src/diffusers/models/unets/unet_2d_condition.py:466]: This PR passes all the tests; so this assumption seems correct.

Comment (tolgacangoz) [reply src/diffusers/models/unets/unet_2d_condition.py:466]: @DN6 This PR passes all the tests; so this assumption seems correct.

**Review by yiyixuxu — APPROVED** [review]
thank you!

**Comment by tolgacangoz** [comment]
Thanks for merging!

**Pull request merged**